include/flang/Lower/PFTBuilder.h

```diff
@@ -115,54 +116,81 @@ static constexpr bool isFunctionLike{common::HasMember<
                   parser::SeparateModuleSubprogram>>};
 
 using LabelSet = llvm::SmallSet<parser::Label, 5>;
-using SymbolLabelMap = std::map<semantics::Symbol *, LabelSet>;
-
-/// Function-like units contain lists of evaluations.  These can be simple
-/// statements or constructs, where a construct contains its own evaluations.
-struct Evaluation {
-  using EvaluationTuple = common::CombineTuples<ActionStmts, OtherStmts,
-                                                ConstructStmts, Constructs>;
-
-  /// Hide non-nullable pointers to the parse-tree node.
-  /// Build type std::variant<const A* const, const B* const, ...>
-  /// from EvaluationTuple type (std::tuple<A, B, ...>).
-  template <typename A>
-  using MakeRefType = const A *const;
-  using EvaluationVariant = common::CombineVariants<
-      common::MapTemplate<MakeRefType, EvaluationTuple>>;
-  template <typename A>
-  constexpr auto visit(A visitor) const {
-    return std::visit(common::visitors{
-                          [&](const auto *p) { return visitor(*p); },
-                          [&](auto &r) { return visitor(r); },
-                      },
-                      u);
+using SymbolLabelMap = llvm::DenseMap<const semantics::Symbol *, LabelSet>;
```

Comment:
It's rather ironic that you jumped through all these hoops to ensure a `const T *const` was not null and immutable and yet "Boom! Like that" `Fortran::semantics::Symbol` appears as a pointer.

Reply:
Yes, I tried to put a `SymbolRef` here, which is its perfect use case, but I got errors and was lazy to dig more. I tried again today, oh boy that was an unpleasant experience, I fought to try defining the related DenseMapInfo, failed, try using a std::map, failed (which makes absolute no sens since it already used in f18 like [here ](https://github.com/flang-compiler/f18/blob/a65b690ef00351bd6bc02593ddbd45fa240e9cbc/include/flang/Evaluate/constant.h#L200).
It simply turned out having Symbol incomplete when putting a SymbolRef in a map is a bad idea... but compilers could not simply tell that ^^.
Anyway, solved !